src/5.0_to_5.1/migrateCalculatedMeasuresToActivePivot/migrateCalculatedMeasuresInDashboards.ts

```diff
@@ -0,0 +1,100 @@
+import {
+  ContentEntry,
+  ContentRecord,
+  CubeName,
+  DashboardState,
+  DataModel,
+  deserializeDashboardState,
+  MdxSelect,
+  serializeDashboardState,
+  WidgetWithQueryState,
+} from "@activeviam/activeui-sdk-5.0";
+import { produce } from "immer";
+import { migrateCalculatedMeasuresInMdx } from "./migrateCalculatedMeasuresInMdx";
+
+interface DashboardsFolder {
+  entry: ContentEntry;
+  children: {
+    thumbnails: ContentRecord;
+    content: ContentRecord;
+    structure: ContentRecord;
+  };
+}
+
+/**
+ * Takes a `ui/dashboards` folder from the content server, removes any calculated member definitions from the MDX and returns a `migratedDashboards` folder.
+ * Also returns the cubeName for each calculated measure found.
+ */
+export const migrateCalculatedMeasuresInDashboards = (
+  dashboards: DashboardsFolder,
+  dataModel: DataModel,
+  namesOfCalculatedMeasurestoMigrate: string[],
+): {
+  cubeNames: { [measureName: string]: CubeName };
+  migratedDashboards: DashboardsFolder;
+} => {
+  const cubeNames: { [measureName: string]: CubeName } = {};
+
+  const migratedDashboards = produce(dashboards, (draft) => {
+    const allDashboards = dashboards.children.content.children ?? {};
```

Comment:
I would go for `dashboardsContent` or `dashboardContents` since we're talking about what's under `ui/dashboards/content`, which holds the actual dashboard states as opposed to `ui/dashboards/structure` which only holds metadata
```suggestion
    const dashboardContents = dashboards.children.content.children ?? {};
```